crypto/hpke/hpke_util.c

```diff
@@ -82,18 +70,27 @@ int ossl_hpke_kdf_expand(EVP_KDF_CTX *kctx,
 int ossl_hpke_labeled_extract(EVP_KDF_CTX *kctx,
                               unsigned char *prk, size_t prklen,
                               const unsigned char *salt, size_t saltlen,
+                              const char *protocol_label,
                               const unsigned char *suiteid, size_t suiteidlen,
                               const char *label,
                               const unsigned char *ikm, size_t ikmlen)
 {
     int ret = 0;
     size_t labeled_ikmlen = 0;
-    unsigned char labeled_ikm[LABELED_EXTRACT_SIZE];
+    unsigned char *labeled_ikm = NULL;
```

Comment:
Another way of doing this is to leave the stack buffer here and only allocate one if needed for large sizes..
Then you just use a unsigned char *pbuffer = labeled_ikm; and use the pbuffer to swap to a malloc is needed.

Reply:
would that be worth the complexity? don't think any ossl_hpke_* will be called in a very tight loop

Reply:
it is not very complex, and these functions get called multiple times.

Reply:
not convinced tbh due to how relatively infrequently HPKE will be used but would be happy to change if additional opinion favours the more complex variant